cheroot/server.py

```diff
@@ -1204,6 +1247,289 @@ def send_headers(self):
+                self.close_connection = True
+            else:
+                # TODO
+                raise NotImplementedError("Only expecting Request object here")
+        except h11.RemoteProtocolError as e:
+            # NEED INFO: Should we adjust the tests to match h11's exception text
+            # Or should we continue to shim like this:
+            if e.args[0] == "bad Content-Length":
+                self.simple_response(e.error_status_hint or 400, "Malformed Content-Length Header.")
+            elif e.args[0] == "illegal request line":
+                self.simple_response(e.error_status_hint or 400, 'Malformed Request-Line')
+            elif e.args[0] == "illegal header line":
+                self.simple_response(e.error_status_hint or 400, 'Illegal header line.')
+            else:
+                self.simple_response(e.error_status_hint or 400, str(e))
+
+    def respond(self):
+        """Call the gateway and write its iterable output."""
+        mrbs = self.server.max_request_body_size
+        if self.chunked_read:
+            self.rfile = ChunkedRFile(self.h_conn, self.conn.rfile, self.conn.wfile, mrbs)
+        else:
+            cl = int(self.inheaders.get(b'content-length', 0))
+            if mrbs and mrbs < cl:
+                if not self.sent_headers:
+                    self.simple_response(
+                        '413 Request Entity Too Large',
+                        'The entity sent with the request exceeds the '
+                        'maximum allowed bytes.',
+                    )
+                return
+            self.rfile = KnownLengthRFile(self.h_conn, self.conn.rfile, self.conn.wfile, cl)
+
+        # client may still be in send body, lets find out and figure out what to do with that
+        # if self.h_conn.their_state == h11.SEND_BODY:
+        if self.h_conn.client_is_waiting_for_100_continue:
+            mini_headers = ()
+            go_ahead = h11.InformationalResponse(status_code=100, headers=mini_headers)
+            bytes_out = self.h_conn.send(go_ahead)
+            self.conn.wfile.write(bytes_out)
+        try:
+            self.server.gateway(self).respond()
+            self.ready and self.ensure_headers_sent()
+        except errors.MaxSizeExceeded as e:
+            self.simple_response(413, "Request Entity Too Large")
+            self.close_connection = True
+
+        # If we haven't sent our end-of-message data, send it now
+        if self.h_conn.our_state is not h11.DONE:
+            bytes_out = self.h_conn.send(h11.EndOfMessage())
+            self.conn.wfile.write(bytes_out)
+
+        while self.h_conn.their_state is h11.SEND_BODY:
+            # empty their buffer ?
+            self._next_event()
+
+        if self.h_conn.their_state is h11.MUST_CLOSE:
+            self.close_connection = True
+
+        # prep for next req cycle if it's available
+        if self.h_conn.our_state == h11.DONE and self.h_conn.their_state == h11.DONE:
```

Comment:
For consistency, use the same style of checking for these constants as earlier in code.
```suggestion
        if self.h_conn.our_state is h11.DONE and self.h_conn.their_state is h11.DONE:
```